### DISCUSSION TITLE
Breadcrumbs is going to be large and messy🥲

### DISCUSSION THREAD

**Discussion by HananoshikaYomaru** [SkepticMystic/breadcrumbs#335] [Ideas]
# To be honest...

I love this plugin very much. I discovered this plugin in September 2021 and at that time breadcrumbs does not have this many functions. I found that the functions were not quite powerful for me because I can mimic most of the functions simply using dvjs so I uninstalled it. But I am honestly impressed by the continuous development of this plugin and recently I installed it again and play around with it. It is truly amazing. It solved the Obsidian structuring problem and I believe it will continue to grow and be more comprehensive and eventually become the ultimate tool for vault structuring.

**But there are some problems with the plugin**. I have noticed there are extremely many settings for the plugin and BC seems to have a diverse development goal. For example, it tries to cater to many alternative hierarchy methods, it tries to cater to many ways of adjustment, it tries to cater to many ways of visualization. Everything needs time and effort to develop but nobody will use every method, to be honest. People only stick to their favourite way of structuring. 

The core of breadcrumbs is a way to build structure and return the structured data. The core of BC should be exposed as an API. Alternative Hierarchies and visualization should be the extension of the plugin, which could be separated to other plugins as well (just my thought), leaving only essential and the most common methods in the plugin. #329 

This could make breadcrumb to be faster as fewer things are loaded and also fewer potential bugs because the project has a more simple structure.  

This could also boost the BC community because people will be able to expand the BC functionalities by themselves, using their favourite ways, and surely they will handle the logic and bugs by themselves. 

I hope this makes sense.

**Comment by HananoshikaYomaru** [comment]
I truly appreciate the plugin. I really want this plugin to be a well-developed, very much bug-free Obsidian super power like dataview. To achieve this, I believe the scope of BC should be restricted and the project structure need to be revised.

**Comment by HEmile** [comment]
I think you make some good point, to be honest! Having Breadcrumbs like this makes it hard to maintain, but also hard to get into. The Breadcrumbs settings page is rather overwhelming, and imho suggest that all alternative hierarchy options are equally supported/useful/recommended. 

In my perfectly honest opinion, the original design of BC is most powerful: Manually having to add typed links is what makes one truly organize their notes instead of relying on other weaker structures for organization. I feel a lot of people now will miss this point because they see the easier to use (at least initially) alt hierarchy options.

**Reply by HananoshikaYomaru** [reply]
I really agree with what you said. Sometimes I just work to make BC to perform perfectly on one hierarchy method (mine is hierarchy notes) and I care nothing about the others. I found the BC's effort in hierarchy notes is not enough. For example, typed link definition in hierarchy notes is not complete (e.g. cannot define next prev relation) so I decide to help on it. I am sure people will do so to their favourite alternative hierarchy. 

I true agree that typed link is a powerful feature provided by BC. It lets people build a graph isolated from other graphs. Unlike the original design of obsidian, when you have too many notes, notes just scatter all over the vault, BC will solve your pain point.
